js/script.js

```diff
@@ -30,39 +34,85 @@ const addTask = () => {
     id: Date.now(),
     title: taskTitle,
     createdAt: formatDate(new Date()),
+    isBeingEdited: false,
   };
-
   tasks.unshift(newTask);
+
   renderTasks();
   $taskTitle.value = "";
   closeForm();
 };
 
-const createTaskCard = (task) => {
-  const taskCard = document.createElement("div");
-  taskCard.className = "task-card";
-
-  const titleElement = document.createElement("p");
-  titleElement.textContent = task.title;
-  taskCard.appendChild(titleElement);
-
-  const createdAtElement = document.createElement("p");
-  createdAtElement.className = "created-at";
-  createdAtElement.textContent = `Created At: ${task.createdAt}`;
-  taskCard.appendChild(createdAtElement);
+const updateTask = (taskId, newTitle) => {
+  const taskIndex = tasks.findIndex((task) => task.id === taskId);
+  if (taskIndex !== -1) {
+    tasks[taskIndex].title = newTitle;
+    tasks[taskIndex].isBeingEdited = false;
+  }
+};
```

Comment:
```suggestion
const updateTask = (tasks, taskId, newTitle) => {
  return tasks.map(task =>
    task.id === taskId
      ? { ...task, title: newTitle, isBeingEdited: false }
      : task
  );
};
```

```
tasks = udpateTask(tasks, taskId, newTitle);
```

Now it becomes a pure function and we are immutably updating our state.